File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/fieldsecurity/cs/retrieveusersharedattributepermissions2.cs
// <snippetretrieveusersharedattributepermissions2>



 #region Check if this user has prvReadPOAA
 // Get the GUID of the current user.
 WhoAmIRequest whoAmI = new WhoAmIRequest();
 Guid userLoggedId = 
     ((WhoAmIResponse)_serviceProxy.Execute(whoAmI)).UserId;
 Console.WriteLine("User logged: " + userLoggedId);

 // Check if this user has prvReadPOAA.
 RetrieveUserPrivilegesRequest userPrivilegesRequest = 
     new RetrieveUserPrivilegesRequest();
 userPrivilegesRequest.UserId = userLoggedId;
 RetrieveUserPrivilegesResponse userPrivilegesResponse =
     (RetrieveUserPrivilegesResponse)_serviceProxy.Execute(userPrivilegesRequest);

 // Fixed the GUID for prvReadPOAA.
 Guid prvReadPOAA = new Guid("{68564CD5-2B2E-11DF-80A6-00137299E1C2}");

 if (userPrivilegesResponse.RolePrivileges.Any(r => r.PrivilegeId.Equals(prvReadPOAA)))
 {
     Console.WriteLine("This user DOES have prvReadPOAA");
 }
 else
 {
     Console.WriteLine("This user DOESN'T have prvReadPOAA");
 }
 Console.WriteLine();
 #endregion Check if this user has prvReadPOAA

// </snippetretrieveusersharedattributepermissions2>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/fieldsecurityvb/vb/retrieveusersharedattributepermissions2.vb
' <snippetretrieveusersharedattributepermissions2>



 '					#Region "Check if this user has prvReadPOAA"
 ' Get the GUID of the current user.
 Dim whoAmI As New WhoAmIRequest()
 Dim userLoggedId As Guid = (CType(_serviceProxy.Execute(whoAmI), 
                             WhoAmIResponse)).UserId
 Console.WriteLine("User logged: " &amp; userLoggedId.ToString())

 ' Check if this user has prvReadPOAA.
 Dim userPrivilegesRequest As New RetrieveUserPrivilegesRequest()
 userPrivilegesRequest.UserId = userLoggedId
 Dim userPrivilegesResponse As RetrieveUserPrivilegesResponse =
     CType(_serviceProxy.Execute(userPrivilegesRequest), RetrieveUserPrivilegesResponse)

 ' Fixed the GUID for prvReadPOAA.
 Dim prvReadPOAA As New Guid("{68564CD5-2B2E-11DF-80A6-00137299E1C2}")

 If userPrivilegesResponse.RolePrivileges.Any(Function(r) r.PrivilegeId.Equals(prvReadPOAA)) Then
     Console.WriteLine("This user DOES have prvReadPOAA")
 Else
     Console.WriteLine("This user DOESN'T have prvReadPOAA")
 End If
 Console.WriteLine()
 '					#End Region ' Check if this user has prvReadPOAA

' </snippetretrieveusersharedattributepermissions2>